### PR TITLE
REF: remove last major pytables state-altering

### PR DESCRIPTION
**Pull request by jbrockmendel**
Mentioned in #30344.

### CODE REVIEW

File: pandas/io/pytables.py [hunk 1]
```diff
@@ -3635,23 +3645,21 @@ def _create_axes(
...
+        """
+        Create and return the axes.
 
+        Parameters
+        ----------
+        axes: a list of the axes in order to create (names or numbers of the axes)
+        obj : DataFrame
+            The object to create axes on.
+        validate: bool, default True
+            Whethr to validate the obj against an existing object already written.
```
Comment (crayxt) [inline pandas/io/pytables.py]: Whether

File: pandas/io/pytables.py [hunk 2]
```diff
@@ -3635,23 +3645,21 @@ def _create_axes(
...
-            nan_rep : a values to use for string column nan_rep
-            encoding : the encoding for string values
-            data_columns : a list of columns that we want to create separate to
-                allow indexing (or True will force all columns)
+        """
+        Create and return the axes.
 
+        Parameters
+        ----------
+        axes: a list of the axes in order to create (names or numbers of the axes)
```
Comment (jreback) [inline pandas/io/pytables.py]: can you edit these to be proper doc-strings

**Review by jreback — CHANGES_REQUESTED** [review]
lgtm,  comments

File: pandas/io/pytables.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/io/pytables.py]: formalized it to the extent I could.  A lot of these docstrings could use improvement

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks

**Pull request merged**